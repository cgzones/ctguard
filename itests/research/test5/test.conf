research {
    rules_file = "rules.xml"
}
