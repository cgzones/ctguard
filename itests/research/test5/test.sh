#!/bin/sh

set -eu

BIN=../../../src/research/ctguard-research
if ! [ -e ${BIN} ]; then
    echo "Could not find binary at '${BIN}'!"
    exit 1
fi

cleanup () {
    rm -f test.output
}

cleanup

chmod 640 rules.xml test.conf

OUTPUT=$(${BIN} --cfg-file test.conf --input -f << EOF
Aug 23 04:59:58 server02 sshd[8268]: Bad protocol version identification '\026\003\001\0018\001' from 118.193.31.181 port 41185
quit
EOF
)
echo "$OUTPUT" > test.output

echo "Comapring expected vs actual output:"
diff -u test.output.expected test.output

cleanup

echo "SUCCESS!"
