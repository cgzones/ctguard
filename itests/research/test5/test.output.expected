ctguard-research

  Enter loginput to see its handling.
   - use 'quit' to exit

>
  Processing 'Aug 23 04:59:58 server02 sshd[8268]: Bad protocol version identification '8' from 118.193.31.181 port 41185' ...
    Format (#0) ...  no format match
      Traits:
        control -> ##false##
        format -> ##unknown##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
      Groups: ####
      Priority: 0
    Processing rules (#1) level 1
    Checking level 1 rule 2 ...regex match|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 2
    Checking child rules (#0) ...
    Processing rules (#0) level 2
    Matching level 2 rule: 0
      Traits:
        control -> ##false##
        format -> ##unknown##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        srcip -> ##118.193.31.181##
      Groups: ####
      Priority: 5
    Checking group rules...
    Processing rules (#1) level 1
    Checking level 1 rule 1 ...group always match|regex match|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 1
    Checking child rules (#0) ...
    Processing rules (#0) level 2
    Matching level 2 rule: 0
      Traits:
        control -> ##false##
        format -> ##unknown##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        srcip -> ##118.193.31.181##
      Groups: ####
      Priority: 7
  End extracting.

Rule matched:
	rule id  : 1
	priority : 7
	message  : Possible exploit attempt.
Would create alert with priority 7.

>
Ending
